lib/ansible/plugins/connection/docker.py

```diff
@@ -106,33 +111,35 @@ def _sanitize_version(version):
 
     def _get_docker_version(self):
 
-        cmd = [self.docker_cmd]
-
-        if self._play_context.docker_extra_args:
-            cmd += self._play_context.docker_extra_args.split(' ')
-
+        cmd = list(self.docker_cmd)
         cmd += ['version']
 
-        cmd_output = subprocess.check_output(cmd)
+        try:
+            cmd_output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
```

Comment:
subprocess.check_output is python 2.7+ and controller side code like connection plugins target python2.6

This module was using subprocess.check_output before this pr, but that needs to be fixed as well
(see https://github.com/ansible/ansible/pull/16841). This pr doesn't need to fix the existing cases, but should avoid adding new use of subprocess.check_output()